app/v2/services/InboundRouterService.scala

```diff
@@ -53,9 +53,9 @@ class InboundRouterServiceImpl @Inject()(inboundRouterConnector: InboundRouterCo
         .post(messageType, message, departureId)
         .map(response => {
           if (is2xx(response.status)) {
-            response.header(LOCATION) match {
+            response.header(MessageIdHeaderKey) match {
               case Some(loc) => Right(MessageId(Utils.lastFragment(loc)))
-              case _         => Left(PersistenceError.UnexpectedError(None))
+              case _         => Left(PersistenceError.UnexpectedError(Some(new Exception("Location header missing from router response"))))
```

Comment:
the error message refers to the location header but the header we check is not the messageId header?

Reply:
Forgot to rename loc variable, its reading the correct value from the header though